engine/src/test/java/org/hibernate/validator/test/cfg/ProgrammaticConstraintDefinitionsTest.java

```diff
@@ -66,6 +69,61 @@ public void safeHtmlProgrammaticDefinition() {
 		doProgrammaticTest( new SafeHtmlDef().whitelistType( SafeHtml.WhiteListType.RELAXED ).baseURI( "http://localhost" ),
 				"<img src='/some/relative/url/image.png' />", 0
 		);
+
+		doProgrammaticTest(
+				new SafeHtmlDef().whitelistType( SafeHtml.WhiteListType.NONE )
+						.additionalTags( new TagDef( "img" )
+								.attributes( "src" )
+						),
+				"<img src='data:image/png;base64,100101' />", 1
```

Comment:
So that's the test failing for me after my changes and I've come to think that this assertion actually is wrong. There should be no violation, as `protocol()` ﻿hasn't been given, hence no restrictions apply for the protocols.

Before my change it would result in a violation as we'd call `addProtocol()` with an empty list of protocols, i.e. all the protocols are disallowed.